schema.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/xeipuuv/gojsonschema"
+)
+
+// Refer to http://json-schema.org/ on how to use JSON Schemas.
+
+const (
+	publicSettingsSchema = `{
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "title": "Custom Script - Public Settings",
+    "type": "object",
+    "properties": {
+        "commandToExecute": {
+            "description": "Command to be executed",
+            "type": "string"
+        },
+        "fileUris": {
+            "description": "List of files to be downloaded",
+            "type": "array",
+            "items": {
+                "type": "string",
+				"format": "uri"
+            }
+        },
+		"timestamp": {
+			"description": "An integer, intended to trigger re-execution of the script when changed",
+			"type":"integer" 
+		}
+    },
+	"additionalProperties": false,
+    "required": ["commandToExecute"]
+}`
+
+	protectedSettingsSchema = `{
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "title": "Custom Script - Protected Settings",
+    "type": "object",
+    "properties": {
+        "commandToExecute": {
+            "description": "Command to be executed",
+            "type": "string"
+        },
+        "storageAccountName": {
+            "description": "Name of the Azure Storage Account (3-24 characters of lowercase letters or digits)",
+            "type": "string",
+			"pattern": "^[a-z0-9]{3,24}$"
+        },
+		"storageAccountKey": {
+            "description": "Key for the Azure Storage Account (a base64 encoded string)",
+            "type": "string",
+			"pattern": "^(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{4})$"
+        }
+    },
+	"additionalProperties": false
+}`
+)
+
+// validateObjectJSON validates the specified json with schemaJSON.
+// If json is empty string, it will be converted into an empty JSON object
+// before being validated.
+func validateObjectJSON(schema *gojsonschema.Schema, json string) error {
+	if json == "" {
```

Comment:
@boumenot because it does not necessarily make it valid. consider empty public settings json. It does not contain `commandToExecute`; so it's not valid and it has to be validated.

However if I pass the empty string to `schema.Validate` it'll error out saying, `EOF` because empty string is not valid JSON at all (`""` is valid json, `null` is valid json, zero bytes isn't).

Therefore instead of passing empty string (which would give me an error like: `expected: object, got: string`, because public settings should be a JSON object), now I convert empty string into `{}` and pass it to validation and it tells me `required field empty: commandToExecute`.